### PR TITLE
Documentation and convenience updates

### PR DESCRIPTION
**Pull request by jtpereyda**
A few changes I recorded as I was installing and debugging that should make it easier for future users.

 - nsscache.conf    - Changed default cache value to 'nssdb' from 'cache' -- 'cache' is no longer a valid option in nsscache.conf.5 or config.py.
 - nsscache.conf.5  - Added ldap_tls_starttls, listed ldap_scope valid options, typo fixes.
 - ldapsource.py    - Added synonyms 'onelevel' and 'subtree' to match existing nslcd.conf conventions.

I think I'm almost out of your hair now. :)

### CODE REVIEW

**Comment by jaqx0r** [comment]
Why is cache no longer a valid option?

**Comment by jaqx0r** [comment]
Or phrased another way, did you know that the nssdb backends is slow and obsolete? The cache backend is recommended for all users.  I noticed many of your reports contain the string 'nssdb' and am curious what is missing in the docs that made you choose it.

File: nsscache.conf [hunk 1]
```diff
@@ -13,7 +13,7 @@
 source = ldap
 
 # Default NSS data cache module name
-cache = cache
+cache = nssdb
```
Comment (jaqx0r) [inline nsscache.conf]: Revert this change.

**Comment by jtpereyda** [comment]
I thought that 'cache' was no longer valid after reading the man page for nsscache.conf, hosted [here](http://linux.die.net/man/5/nsscache.conf). About the cache option, it says:
> "Valid options: nssdb".
Although, looking at the source in this repo, it looks like cache is an option.

**Comment by jaqx0r** [comment]
That manpage is out of date, then; unfortunately it doesn't render the
version string at the top of of the manpage source.  The project at GitHub
should be the canonical reference, which says that the nssdb option is
"deprecated in favour of cache."

**Comment by jaqx0r** [comment]
Please revert the config change as requested and I'll merge.

**Comment by jtpereyda** [comment]
Reverted. Sorry for the delay.

**Comment by jaqx0r** [comment]
Thanks!
